src/kernel/include/pgtbl.h

```diff
@@ -56,527 +94,45 @@ struct tlb_quiescence {
-                                                  PGTBL_DEPTH, &accum);
-	orig_v = (u32_t)(pte->next);
-	if (!(orig_v & PGTBL_PRESENT)) return -EEXIST;
-	if (orig_v & PGTBL_COSFRAME) return -EPERM;
-
-
-	ret = __pgtbl_update_leaf(pte, (void *)((liv_id << PGTBL_PAGEIDX_SHIFT) | PGTBL_QUIESCENCE), orig_v);
-	if (ret) cos_throw(done, ret);
-
-	/* decrement ref cnt on the frame. */
-	ret = retypetbl_deref((void *)(orig_v & PGTBL_FRAME_MASK));
-	if (ret) cos_throw(done, ret);
-
-done:
-	return ret;
-}
-
-/* NOTE: This just removes the mapping. NO liveness tracking! TLB
- * flush should be taken care of separately (and carefully). */
-static int
-pgtbl_mapping_del_direct(pgtbl_t pt, u32_t addr)
-{
-	unsigned long accum = 0, *pte = NULL;
-
-	assert(pt);
-	assert((PGTBL_FLAG_MASK & addr) == 0);
-
-	return __pgtbl_expandn(pt, addr >> PGTBL_PAGEIDX_SHIFT, PGTBL_DEPTH + 1, &accum, &pte, NULL);
-}
-
-static void *
-pgtbl_lkup_lvl(pgtbl_t pt, u32_t addr, u32_t *flags, u32_t start_lvl, u32_t end_lvl)
-{
-	return __pgtbl_lkupani((pgtbl_t)((unsigned long)pt | PGTBL_PRESENT), addr >> PGTBL_PAGEIDX_SHIFT, start_lvl,
-	                       end_lvl, flags);
-}
-
-static int
-pgtbl_ispresent(u32_t flags)
-{
-	return flags & (PGTBL_PRESENT | PGTBL_COSFRAME);
-}
-
-static unsigned long *
-pgtbl_lkup(pgtbl_t pt, u32_t addr, u32_t *flags)
-{
-	void *ret;
-
-	ret = __pgtbl_lkupan((pgtbl_t)((unsigned long)pt | PGTBL_PRESENT), addr >> PGTBL_PAGEIDX_SHIFT, PGTBL_DEPTH + 1,
-	                     flags);
-	if (!pgtbl_ispresent(*flags)) return NULL;
-	return ret;
-}
-
-/* Return the pointer of the pte.  */
-static unsigned long *
-pgtbl_lkup_pte(pgtbl_t pt, u32_t addr, u32_t *flags)
-{
-	return __pgtbl_lkupan((pgtbl_t)((unsigned long)pt | PGTBL_PRESENT), addr >> PGTBL_PAGEIDX_SHIFT, PGTBL_DEPTH,
-	                      flags);
-}
-
-/* FIXME: remove this function.  Why do we need a paddr lookup??? */
-static paddr_t
-pgtbl_lookup(pgtbl_t pt, u32_t addr, u32_t *flags)
-{
-	unsigned long *ret = pgtbl_lkup(pt, addr, flags);
-	if (!ret) return (paddr_t)NULL;
-	return (paddr_t)chal_va2pa(ret);
-}
-
-static int
-pgtbl_get_cosframe(pgtbl_t pt, vaddr_t frame_addr, paddr_t *cosframe)
-{
-	u32_t          flags;
-	unsigned long *pte;
-	paddr_t        v;
-
-	pte = pgtbl_lkup_pte(pt, frame_addr, &flags);
-	if (!pte) return -EINVAL;
-
-	v = *pte;
-	if (!(v & PGTBL_COSFRAME)) return -EINVAL;
-
-	*cosframe = v & PGTBL_FRAME_MASK;
-
-	return 0;
-}
+int pgtbl_cosframe_add(pgtbl_t pt, u32_t addr, u32_t page, u32_t flags);
+int pgtbl_mapping_add(pgtbl_t pt, u32_t addr, u32_t page, u32_t flags);
+int pgtbl_mapping_mod(pgtbl_t pt, u32_t addr, u32_t flags, u32_t *prevflags);
+int pgtbl_mapping_del(pgtbl_t pt, u32_t addr, u32_t liv_id);
+int pgtbl_mapping_del_direct(pgtbl_t pt, u32_t addr);
+void *pgtbl_lkup_lvl(pgtbl_t pt, u32_t addr, u32_t *flags, u32_t start_lvl, u32_t end_lvl);
+int pgtbl_ispresent(u32_t flags);
+unsigned long *pgtbl_lkup(pgtbl_t pt, u32_t addr, u32_t *flags);
```

Comment:
You might want to align declarations. 
eg.
```
int                        pgtbl_xxxx(xxxx);
unsigned long *pgtbl_yyyy(yyyy);
```

Reply:
wow.. that is not how I intended it to be.. the function names should have started at the same column.

Reply:
ok